app/src/main/java/org/dhis2/usescases/programEventDetail/eventList/EventListViewModel.kt

```diff
@@ -41,13 +45,17 @@ class EventListViewModel(
                     .map { pagingData ->
                         pagingData.map { event ->
                             withContext(dispatchers.io()) {
+                                _displayOrgUnitName.postValue(
+                                    event.program()?.let { program ->
+                                        eventRepository.displayOrganisationUnit(program)
+                                    } ?: true,
+                                )
+
                                 val eventModel = mapper.eventToEventViewModel(event)
                                 cardMapper.map(
                                     event = eventModel,
                                     editable = eventRepository.isEventEditable(event.uid()),
-                                    displayOrgUnit = event.program()?.let { program ->
-                                        eventRepository.displayOrganisationUnit(program)
-                                    } ?: true,
+                                    displayOrgUnit = displayOrgUnitName.value ?: true,
```

Comment:
What do you think about create a test in `EventListViewModel` to assert the `eventModel` object and check `displayOrgUnit` property has expected value?